lib/dfe/analytics/entity_table_check_job.rb

```diff
@@ -27,16 +27,47 @@ def perform
 
       def entity_table_check_data(model)
         checksum_calculated_at = Time.now.in_time_zone(TIME_ZONE).iso8601(6)
+        adapter_name = ActiveRecord::Base.connection.adapter_name.downcase
+        row_count, checksum = fetch_checksum_data(model, checksum_calculated_at, adapter_name)
+
+        {
+          row_count: row_count,
+          checksum: checksum,
+          checksum_calculated_at: checksum_calculated_at
+        }
+      end
+
+      def fetch_checksum_data(model, checksum_calculated_at, adapter_name)
+        if adapter_name == 'postgresql'
+          fetch_postgresql_checksum_data(model, checksum_calculated_at)
+        else
+          fetch_generic_checksum_data(model, checksum_calculated_at)
+        end
+      end
+
+      def fetch_postgresql_checksum_data(model, checksum_calculated_at)
+        checksum_sql_query = <<-SQL
+          SELECT COUNT(*) as row_count,
+            MD5(STRING_AGG(CHECKSUM_TABLE.ID, '')) as checksum
+          FROM (
+            SELECT #{model.table_name}.id::TEXT as ID
+            FROM #{model.table_name}
+            WHERE #{model.table_name}.updated_at < '#{checksum_calculated_at}'
+            ORDER BY #{model.table_name}.updated_at ASC
+          ) CHECKSUM_TABLE
+        SQL
+
+        result = ActiveRecord::Base.connection.execute(checksum_sql_query).first
+        [result['row_count'].to_i, result['checksum']]
+      end
+
+      def fetch_generic_checksum_data(model, checksum_calculated_at)
         table_ids =
           model
           .where('updated_at < ?', Time.parse(checksum_calculated_at))
           .order(updated_at: :asc)
           .pluck(:id)
-        {
-          row_count: table_ids.count,
-          checksum: Digest::SHA256.hexdigest(table_ids.join),
-          checksum_calculated_at: checksum_calculated_at
-        }
+        [table_ids.count, Digest::SHA256.hexdigest(table_ids.join)]
```

Comment:
This should be MD5 now rather than SHA256 to keep both versions consistent. I believe the method is: Digest::MD5.